### PR TITLE
fix(tabs): home and end navigation

### PR DESCRIPTION
**Pull request by m4thieulavoie**
# Pull Request

## 📖 Description

This PR makes sure to focus on the first/last item that is _focusable_ on `home`/`end` key presses. I didn't add tests as there's still a todo for keyboard navigation tests in the test suite.

### 🎫 Issues

https://github.com/microsoft/fast/issues/6275

### General

- [x] I have included a change request file using `$ yarn change`
- [ ] I have added tests for my changes.
- [x] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific

- [ ] I have added a new component
- [ ] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

### CODE REVIEW

**Comment by chrisdholt** [comment]
Thanks @m4thieulavoie! 

> I didn't add tests as there's still a todo for keyboard navigation tests in the test suite.

No worries, this came from @radium-v moving us to playwright and I'm betting he wrote a test which exposed this :)

File: packages/web-components/fast-foundation/src/tabs/tabs.ts [hunk 1]
```diff
@@ -364,6 +364,22 @@ export class FASTTabs extends FASTElement {
             this.tabs.length - 1,
             this.activeTabIndex + adjustment
         );
+
+        if (this.activeTabIndex > this.prevActiveTabIndex) {
+            while (
```
Comment (scomea) [inline packages/web-components/fast-foundation/src/tabs/tabs.ts:369]: Can't these "while" loops get out of thre range of valid indexes if there are no focusable tabs found as you increment/decrement?

Comment (scomea) [reply packages/web-components/fast-foundation/src/tabs/tabs.ts:369]: Ok, nvm, I see it now.

**Comment by scomea** [comment]
Smoked, works as advertised.

**Review by scomea — APPROVED** [review]

**Review by radium-v — CHANGES_REQUESTED** [review]
I think this may be more efficient if we find the first and last focusable indices by filtering the tabs (this same method could also be applied to the `adjustForward` and `adjustBackward` functions):

```ts
    public adjust(adjustment: number): void {
        const focusableTabs = this.tabs.filter(t => !this.isDisabledElement(t));
        const currentActiveTabIndex = focusableTabs.indexOf(this.activetab);
        
        const nextTabIndex = limit(
            0,
            focusableTabs.length - 1,
            currentActiveTabIndex + adjustment
        );
        
        // the index of the next focusable tab within the context of all available tabs
        const nextIndex = this.tabs.indexOf(focusableTabs[nextTabIndex]);

        if (nextIndex > -1) {
            this.moveToTabByIndex(this.tabs, nextIndex);
        }
    }
```

The `limit` function is provided by the `@microsoft/fast-web-utilities` package.

**Comment by m4thieulavoie** [comment]
> I think this may be more efficient if we find the first and last focusable indices by filtering the tabs (this same method could also be applied to the `adjustForward` and `adjustBackward` functions):
> 
> ```ts
>     public adjust(adjustment: number): void {
>         const focusableTabs = this.tabs.filter(t => !this.isDisabledElement(t));
>         const currentActiveTabIndex = focusableTabs.indexOf(this.activetab);
>         
>         const nextTabIndex = limit(
>             0,
>             focusableTabs.length - 1,
>             currentActiveTabIndex + adjustment
>         );
>         
>         // the index of the next focusable tab within the context of all available tabs
>         const nextIndex = this.tabs.indexOf(focusableTabs[nextTabIndex]);
> 
>         if (nextIndex > -1) {
>             this.moveToTabByIndex(this.tabs, nextIndex);
>         }
>     }
> ```
> 
> The `limit` function is provided by the `@microsoft/fast-web-utilities` package.

Done! Turns out we can't easily apply that logic to `adjustForward` and `adjustBackward` though. Arrow navigation handles wrapping whereas it's of course not the case for home/end keypresses.

**Comment by radium-v** [comment]
> Arrow navigation handles wrapping whereas it's of course not the case for home/end keypresses.

@m4thieulavoie the `wrapInBounds` function from `@microsoft/fast-web-utilities` should be able to handle `adjustBackward` and `adjustForward` instead of `limit`

**Review by radium-v — APPROVED** [review]

**Review by chrisdholt — APPROVED** [review]
Thanks @m4thieulavoie!

File: change/@microsoft-fast-foundation-dc08ca7b-be41-4f42-a66c-c0874376b4a2.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "patch",
+  "comment": "fix(tabs): home and end navigation",
+  "packageName": "@microsoft/fast-foundation",
+  "email": "mathieulavoie94@gmail.com",
+  "dependentChangeType": "patch"
```
Comment (radium-v) [inline change/@microsoft-fast-foundation-dc08ca7b-be41-4f42-a66c-c0874376b4a2.json:6]:
```suggestion
  "type": "prerelease",
  "comment": "fix(tabs): home and end navigation",
  "packageName": "@microsoft/fast-foundation",
  "email": "mathieulavoie94@gmail.com",
  "dependentChangeType": "prerelease"
```

**Comment by chrisdholt** [comment]
@m4thieulavoie would you mind creating a cherry picked PR against `archives/fast-element-1` for this?

**Pull request merged**